### ISSUE TITLE
Postgresql dbhost directive

### ISSUE THREAD

**Issue by seannaswell** [zotonic/zotonic#2176]
Are the scripts used to startup Zotonic referencing the `{dbhost, "xx.xx.xx.xx"}` directive found in .zotonic/zotonic.config and site/zotonic_site.config? 

Both of those files have a non-local IP address where the Postgres database is located, and the login credentials are entered. When executing `psql -h 10.0.0.10 -U zotonic` a connection to the database is made. However, when starting zotonic the following error appears, after which every other module fails to start due to missing dependencies: 

`17:43:31.732 [info] z_module_manager:1275 [zotonic_site_status] Skipping schema for mod_base as there is no database connection.`

Is it possible that the `psql` command is being issued by startup scripts without referencing the dbhost directive?

**Comment by seannaswell** [comment]
FYI- After setting the [PGHOST environment variable](https://www.postgresql.org/docs/9.6/libpq-envars.html) to the database host IP, executing `psql -U zotonic` establishes a database connection successfully, without requiring specifying the host. 

Therefore, any `psql` command issued by startup scripts will transparently connect to the remote Postgres server rather than local unix sockets. However, Zotonic still fails to establish a database connection when starting.

**Comment by karlsson** [comment]
It looks to me as the log record is of type info, so not a real error?
I get the same info printout but the status site comes up fine anyway:
```13:27:16.877 [info] z_sites_manager:679 Starting site zotonic_site_status
13:27:17.046 [info] z_module_manager:1275 [zotonic_site_status] Skipping schema for mod_base as there is no database connection.
13:27:17.155 [info] z_site_startup:77 [zotonic_site_status] info @ z_site_startup:77  Site zotonic_site_status started, modules loaded
```

**Comment by mworrell** [comment]
Indeed, this `as there is no database connection` is confusing.
It should say something like `as this site does not use a database ('nodb')`

**Comment by karlsson** [comment]
I had some problems with legacy sites and modules not loading because of failing dependency to mod_wires that was not started. Adding `{install_modules, [mod_wires]},` to the zotonic_site.config fixed that problem.
When creating new sites from web gui or using addsite command this problem does not exist for some reason. It seems that they use mod_wires too.

**Comment by mworrell** [comment]
Yes, `mod_wires` is added as a dependency and activated for new sites.

**Comment by mworrell** [comment]
@seannaswell do you suggest to have `PGHOST` (and friends) overrule the definitions in `zotonic.config`?

**Comment by seannaswell** [comment]
@karlsson Yes, I see the message is labelled as [info], and noticed that it appeared after initially compiling and starting zotonic, before any sites are created. It does seem rewording that message would be helpful.

However, in my case that message was followed by errors saying modules didn't start due to failed dependency checks, because of no database connection. I should have posted the other messages instead, thanks for pointing that out.

**Comment by seannaswell** [comment]
@mworrell It seems defining the `dbhost` DNS/IP in `zotonic.config` is easier, as `dbpassword` needs to be set anyway. 

Defining `PGHOST` in `.bashrc` was simply a test to see if the `psql` command was being issued somewhere without referencing the `dbhost` directive.

Currently testing the database connection issue, and expect to find it is _not_ zotonic related...

**Comment by seannaswell** [comment]
@mworrell Actually, you may be correct.

**Comment by seannaswell** [comment]
Okay, problem solved.

FIrst, I upgraded to version 11.5 of postgresql-server on the database server, and postgresql-client on the machine running zotonic. Then, magically, all connectivity issues resolved themselves.

I removed the `PGHOST` directive in `.bashrc`, and zotonic used the IP address specified in the `zotonic_site.config` file, as expected. 

Sorry for the noise, and thanks for the assistance.
